### PR TITLE
BUG FIX: groupby.last() no longer changes np arrays with pd.NA to dtype Int64

### LINKED ISSUE #46409: BUG: grouped.last() will sometimes turn a boolean column into Int64
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame(
    {
        'id': [1, 2, 3, 4],
        'test': [True, pd.NA, pd.NA, False]
    }
).convert_dtypes()

grouped = df.groupby('id')
bad = grouped.last()
assert bad.test.dtype == pd.BooleanDtype() # fails
```

Latest master outputs:
```
In [3]: bad
Out[3]:
    test
id
1      1
2   <NA>
3   <NA>
4      0
```
```


### Issue Description

On the latest master this returns an Int64 column.

```
    test
id
1      1
2   <NA>
3   <NA>
4      0
```
I checked 1.4.0 and it properly returns the boolean dtype.
```
     test
id
1    True
2    <NA>
3    <NA>
4   False
```

What is weird is that changing 3. to True/False will give the proper dtype.

### Expected Behavior

Retain the boolean dtype from `df.test`.

### Installed Versions

INSTALLED VERSIONS
------------------
commit           : 663147edd35bc3e0362f7d637c8d5f5e597f961b
python           : 3.10.0.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.16.12-arch1-1
Version          : #1 SMP PREEMPT Wed, 02 Mar 2022 12:22:51 +0000
machine          : x86_64
processor        :
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.0.dev0+545.g663147edd3
numpy            : 1.21.5
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 58.5.3
Cython           : 0.29.28
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.7.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : 2.0.1
matplotlib       : None
numba            : 0.55.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 8.0.0.dev230+gb2ae3d74d
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.0
snappy           : None
sqlalchemy       : 2.0.0b1
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

### PR DESCRIPTION
**Pull request by sydneyeh**
- [ X] closes #46409  
- [X ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [X ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -457,7 +457,7 @@ Performance improvements
 .. _whatsnew_150.bug_fixes:
 
 Bug fixes
-~~~~~~~~~
+- Bug in :func:`_call_cython_op` that caused numpy arrays ordered with last with pd.NA to have dytype:Int64 type rather than dtype:Bool (:issue:`46409`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.5.0.rst:460]: can you describe this using a user-facing function

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -477,12 +477,12 @@ Datetimelike
 
 Timedelta
 ^^^^^^^^^
-- Bug in :func:`astype_nansafe` astype("timedelta64[ns]") fails when np.nan is included (:issue:`45798`)
+- Bug FIXED in :func:`astype_nansafe` astype("timedelta64[ns]") fails when np.nan is included (:issue:`45798`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.5.0.rst:480]: i dont think this change is necessary

File: pandas/core/groupby/ops.py [hunk 3]
```diff
@@ -571,6 +571,11 @@ def _call_cython_op(
                     **kwargs,
                 )
 
+        if self.how == 'last' and dtype == bool:
+            for val in result:
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py:575]: can you do this without a for loop?

**Comment by sydneyeh** [comment]
@github-actions pre-commit

**Comment by sydneyeh** [comment]
@github-actions pre-commit

**Pull request closed without merging**